src/array.jl

```diff
@@ -193,3 +193,13 @@ function getindex(A::CatOrdArray, i::Int)
 end
 
 levels!(A::CatOrdArray, newlevels::Vector) = _levels!(A, newlevels)
+
+Base.push!{T, R}(A::CategoricalArray{T, 1, R}, item::T) = (push!(A.refs, get!(A.pool, item)); return A)
```

Comment:
Better use the `CatOrdArray` alias as elsewhere, rather than duplicating definitions.